File: tests/fixtures/go/sample.go
package core

import (
	"context"
	"fmt"
)

// Person represents a person entity.
type Person struct {
	Name string
	Age  int
}

// NewPerson creates a new Person.
func NewPerson(name string, age int) *Person {
	return &Person{Name: name, Age: age}
}

// Greet returns a greeting string.
func (p *Person) Greet() string {
	return fmt.Sprintf("Hi, I'm %s", p.Name)
}

// Repository is a generic repository interface.
type Repository[T any] interface {
	GetByID(ctx context.Context, id int) (T, error)
	Save(ctx context.Context, entity T) error
}

// Status represents a status enum.
type Status int

const (
	StatusActive   Status = iota
	StatusInactive
	StatusPending
)

// Point is a 2D point.
type Point struct {
	X, Y int
}

File: tests/fixtures/cs/sample.cs
using System;
using System.Collections.Generic;

namespace MyApp.Core {
    /// <summary>
    /// Represents a person entity.
    /// </summary>
    [Serializable]
    public class Person {
        public string Name { get; set; }
        public int Age { get; set; }

        public Person(string name, int age) {
            Name = name;
            Age = age;
        }

        public async System.Threading.Tasks.Task<Result> ProcessAsync(
            int id,
            string data,
            System.Threading.CancellationToken ct) {
            return await DoWork(id, data, ct);
        }

        public override string ToString() {
            return Name + " (" + Age + ")";
        }
    }

    public interface IRepository<T> {
        System.Threading.Tasks.Task<T> GetByIdAsync(int id);
        System.Threading.Tasks.Task SaveAsync(T entity);
    }

    public enum Status {
        Active,
        Inactive,
        Pending
    }

    public record PersonDto(string Name, int Age);

    public static class PersonFactory {
        public static Person Create(string name, int age) {
            return new Person(name, age);
        }
    }
}
